docs/docs/30-how-to-guides/20-managing-credentials.md

```diff
@@ -122,3 +122,307 @@ It is important to understand the security implications of this feature. Any
+to the required ECR repositories.
+:::
+
+:::caution
+This method of authentication is a "lowest common denominator" approach that
+will work regardless of where Kargo is deployed. i.e. If running Kargo outside
+of EKS, this method will still work.
+
+If running Kargo within EKS, you may wish to consider using EKS Pod Identity
+instead.
+:::
+
+#### EKS Pod Identity
+
+If Kargo is deployed within an EKS cluster, it is possible to use
+[EKS Pod Identity](https://docs.aws.amazon.com/eks/latest/userguide/pod-identities.html)
+to authenticate to ECR repositories. This option eliminates the need to store
+credentials in a `Secret` resource.
+
+First, follow
+[this overview](https://docs.aws.amazon.com/eks/latest/userguide/pod-identities.html#pod-id-setup-overview)
+to set up EKS Pod Identity in your EKS cluster and assign an IAM role to the
+`kargo-controller` `ServiceAccount` within the `Namespace` to which Kargo is (or
+will be) installed.
+
+At this point, an IAM role will be associated with the Kargo _controller_,
+however, that controller acts on behalf of multiple Kargo projects, each of
+which may require access to _different_ ECR repositories. To account for this,
+when Kargo attempts to access an ECR repository on behalf of a specific project,
+it will first attempt to
+[assume an IAM role](https://docs.aws.amazon.com/STS/latest/APIReference/API_AssumeRole.html)
+specific to that project. The name of the role it attempts to assume will _always_
+be of the form `kargo-project-<project name>`. It is this role that should be
+granted read-only access to applicable ECR repositories.
+
+:::info
+The name of the IAM role associated with each Kargo project is deliberately
+not configurable to prevent project admins from attempting to coerce Kargo into
+assuming arbitrary IAM roles.
+:::
+
+Once Kargo is able to assume the appropriate IAM role for a given project, it
+will follow a process similar to that described in the previous section to
+obtain a token that is valid for 12 hours and cached for 10.
+
+:::caution
+Following the principle of least privilege, the IAM role associated with the
+`kargo-controller` `ServiceAccount` should be limited only to the ability to
+assume project-specific IAM roles. Project-specific IAM roles should be limited
+only to read-only access to the applicable ECR repositories.
+:::
+
+### Google Artifact Registry
+
+:::note
+Google Container Registry (GCR) has been deprecated in favor of Google Artifact
+Registry. For authentication to repositories with legacy GCR URLs, the same
+options outlined here may be applied.
+:::
+
+#### Long-Lived Credentials
+
+Google Artifact Registry does not _directly_ support long-lived credentials,
+however, a Google service account key
+[can be exchanged for a token](https://cloud.google.com/artifact-registry/docs/docker/authentication#token)
+that is valid for 60 minutes. Kargo can seamlessly make this exchange and will
```

Comment:
> Google Artifact Registry does not _directly_ support long-lived credentials,

Actually, unlike ECR, GAR **_can_** be accessed with a long-lived service account key as a password, so we should change this wording. Credentials look like:

username: `_json_key_base64`
password: base64 encoded of some json like:

```json
{
  "type": "service_account",
  "project_id": "myproject",
  "private_key_id": "xxxxxx",
  "private_key": "-----BEGIN PRIVATE KEY-----\nyyyyyyy==\n-----END PRIVATE KEY-----\n",
  "client_email": "foo@myproject.iam.gserviceaccount.com",
  "client_id": "12345",
  "auth_uri": "https://accounts.google.com/o/oauth2/auth",
  "token_uri": "https://oauth2.googleapis.com/token",
  "auth_provider_x509_cert_url": "https://www.googleapis.com/oauth2/v1/certs",
  "client_x509_cert_url": "https://www.googleapis.com/robot/v1/metadata/x509/foo%40myproject.iam.gserviceaccount.com"
}
```

This is documented at https://cloud.google.com/artifact-registry/docs/docker/authentication#json-key,

We can point them there if they prefer/need to use long-lived credentials and explain that GAR supports two methods of accessing private repos.

TBH, I'm not sure what benefits "Access Token" auth has over "Service Account Key" since Access Token would still need a long lived service account credential anyways.

Reply:
🤦‍♂️ Embarrassed that I missed that part of the docs, and also question the value of access tokens under these circumstances. I will document this as a third option for authenticating to GAR. For my own edification, I might also research if/why exchanging a service account key for an access token might have any benefit over using the key directly.

Reply:
Addressed. Directly using the service account token is now listed as an option that works, but which Google discourages.

I have also posted a question about that recommendation on security.stackexchange.com.

https://security.stackexchange.com/questions/277163/why-use-access-token-for-google-artifact-registry-access

Reply:
I have much more clarity about this now, and it would have been obvious if I'd re-read my own code after the inital confusion had set in...

I think I allowed the fact that `_json_key_base64` and a base64-encoded service account key _can_ be used as user/pass in basic auth to the registry to mislead me into believing that `gcloud auth print-access-token` (or programmatic equivalent) was using the service account key _in the same way_ -- which is totally not the case.

The service account key is actually used to sign a JWT and it is the JWT that is exchanged for an access token -- meaning that when you request an access token, the service account key does not at any point traverse the wire.

It's right here:

https://github.com/akuity/kargo/blob/088fc4d358d2ed69879e3133d753ea50e1d12a65/internal/credentials/gar/service_account_key.go#L105-L113

This being the case, the benefits of obtaining an access token over using the service account key to access the registry directly are more obvious.

Two action items for me:

1. Without going into exhaustive detail, understanding this much better now compels me to somewhat strengthen my caution against the most direct approach. I will amend this PR accordingly.

2. While researching, I noticed that when requesting the access token, I am requesting a read/write scope, which is quite unnecessary for how Kargo interacts with a registry, which is always in a read-only fashion. I will fix this in a separate PR.

Reply:
Docs amended in b94cb5236a3c3737d46876b6eaea9aa0300fcfa6